Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Commands/Commands.cs

```diff
@@ -130,66 +131,140 @@ public static Command CutActionsOrBindings()
             return (in InputActionsEditorState state) =>
             {
                 CopyPasteHelper.Cut(state);
-                return state.selectionType == SelectionType.Action ?
-                    DeleteAction(state.selectedActionMapIndex, Selectors.GetSelectedAction(state)?.wrappedProperty.FindPropertyRelative(nameof(InputAction.m_Name)).stringValue).Invoke(state)
-                    : DeleteBinding(state.selectedActionMapIndex, state.selectedBindingIndex).Invoke(state);
+                return state.CutActionOrBinding();
             };
         }
 
-        public static Command PasteActionMaps()
+        public static Command PasteActionMaps(IReadOnlyList<IPasteListener> pasteListeners)
         {
             return (in InputActionsEditorState state) =>
             {
-                var lastPastedElement = CopyPasteHelper.PasteActionMapsFromClipboard(state);
+                var newIndex = -99;
+                if (state.GetCutElements() != null && state.GetCutElements().Any())
+                    newIndex = CopyPasteHelper.DeleteCutElements(state);
+                else
+                {
+                    foreach (var pasteListener in pasteListeners)
+                        pasteListener.OnPaste(state);
+                }
+                var lastPastedElement = CopyPasteHelper.PasteActionMapsFromClipboard(state.With(selectedActionMapIndex: newIndex >= -1 ? newIndex : state.selectedActionMapIndex));
```

Comment:
What is `newIndex >= -1 ? newIndex : state.selectedActionMapIndex));` For what cases is it expected to uses state.selectedActionMapINdex? Its there some use of negative indices I am not understanding?

Reply:
same as above (-1 is valid here if the paste of an item happens at the beginning of a list (CopyPasteHelper always adds +1 to insert the item after the index provided))